content/docs/concepts/components_description.md

```diff
@@ -0,0 +1,202 @@
++++
+title = "NSM components"
+weight = 4
+summary = "Network Service Mesh (NSM): The Hybrid/Multi-cloud IP Service Mesh"
++++
+
+## NSM Services
+1. **NetworkService** is used for requesting and closing network connections between clients and endpoints.<br/>
+**Proto File**: https://github.com/networkservicemesh/api/blob/main/pkg/api/networkservice/networkservice.proto
+```proto
+service NetworkService {
+    rpc Request(NetworkServiceRequest) returns (connection.Connection);
+    rpc Close(connection.Connection) returns (google.protobuf.Empty);
+}
+```
+
+2. **Registry** is used for registering and unregistering **Network Services** and **Network Service Endpoints** in NSM, as well as for searching.<br/>
+**Proto File**: https://github.com/networkservicemesh/api/blob/main/pkg/api/registry/registry.proto
+```proto
+service NetworkServiceEndpointRegistry {
+    rpc Register (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
+    rpc Find (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpointResponse);
+    rpc Unregister (NetworkServiceEndpoint) returns (google.protobuf.Empty);
+}
+
+service NetworkServiceRegistry {
+    rpc Register (NetworkService) returns (NetworkService);
+    rpc Find (NetworkServiceQuery) returns (stream NetworkServiceResponse);
+    rpc Unregister (NetworkService) returns (google.protobuf.Empty);
+}
+```
+
+3. **MonitorConnections** service is used for monitoring or searching connections with a specific selector.<br/>
+**Proto File**: https://github.com/networkservicemesh/api/blob/main/pkg/api/networkservice/connection.proto
+```proto
+service MonitorConnection {
+    rpc MonitorConnections(MonitorScopeSelector) returns (stream ConnectionEvent);
+}
+```
+
+## Network Service Manager (NSMgr)
+**Network Service Manager** is one of the key components of NSM, which is responsible for discovering **Network Services** and **Network Service Endpoints** and processing requests from clients. It must be located on the same machine as the NSM client in order to provide it with a connection to the NSM network. It can also serve as a registry if there is no real registry.
+
+**Image**: https://github.com/networkservicemesh/cmd-nsmgr/pkgs/container/cmd-nsmgr
+
+**Implements services**:
+1. NetworkService
+2. Registry
+3. MonitorConnection
+
+**Responsibilities**:
+1. NetworkService discovery
+2. Forwarder discovery
+3. Storage for valid connections from clients
+4. Provides access to the network namespace of the clients and endpoints for the forwarder
+5. Can serve as a registry (optional) 
+
+## Forwarder VPP
```

Comment:
I would suggest making one **Forwarder** header, and include

-  Forwarder VPP

-  Forwarder OVS